extension/indexer/tx_indexer.go

```diff
@@ -0,0 +1,118 @@
+// Copyright (C) 2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package indexer
+
+import (
+	"context"
+	"encoding/binary"
+	"errors"
+
+	"github.com/ava-labs/avalanchego/database"
+	"github.com/ava-labs/avalanchego/ids"
+
+	"github.com/ava-labs/hypersdk/chain"
+	"github.com/ava-labs/hypersdk/consts"
+	"github.com/ava-labs/hypersdk/fees"
+)
+
+var (
+	_ AcceptedSubscriber = (*txIndexer)(nil)
+	_ AcceptedSubscriber = (*noopTxIndexer)(nil)
+)
+
+var (
+	failureByte = byte(0x0)
+	successByte = byte(0x1)
+)
+
+type TxIndexer interface {
+	Accepted(ctx context.Context, blk *chain.StatelessBlock) error
```

Comment:
Probably should just replace this line w/ an embed for the `AcceptedSubscriber` interface it since that's the interface we're using to get access to accepted blocks

Reply:
Done